Tools/test_shard_filter.py

```
@@ -337,50 +373,48 @@ def cmd_generate():
 
-    class_counts = extract_classes(tests)
-    print(f"Discovered {len(tests)} tests in {len(class_counts)} classes, distributing across {total} shards", file=sys.stderr)
+    group_counts = extract_groups(tests)
+    print(f"Discovered {len(tests)} tests in {len(group_counts)} groups, distributing across {total} shards", file=sys.stderr)
 
     os.makedirs(output_dir, exist_ok=True)
 
-    def class_weight(cls):
-        multiplier = WEIGHT_OVERRIDES.get(cls, 1.0)
-        return class_counts[cls] * multiplier
+    def group_weight(group):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `group_weight`

(ANN202)